web-wallet/src/lib/stores/__tests__/networkStore.spec.js

```diff
@@ -0,0 +1,62 @@
+import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
+import { get } from "svelte/store";
+
+import { Network } from "$lib/vendor/w3sper.js/src/mod";
+
+describe("NetworkStore", async () => {
+  const blockHeight = 999_888_777n;
+  const connectSpy = vi.spyOn(Network.prototype, "connect");
+  const disconnectSpy = vi.spyOn(Network.prototype, "disconnect");
+  const blockHeightSpy = vi
+    .spyOn(Network.prototype, "blockHeight", "get")
+    .mockResolvedValue(blockHeight);
+
+  afterEach(() => {
+    connectSpy.mockClear();
+    disconnectSpy.mockClear();
+  });
+
+  afterAll(() => {
+    connectSpy.mockRestore();
+    disconnectSpy.mockRestore();
+    blockHeightSpy.mockRestore();
+  });
+
+  it("should expose methods to connect and disconnect from the network and its connection status", async () => {
```

Comment:
Wondering if we should have this split into 2 (or more) units – like expose method to connect, expose method to disconnect, and one to ensure that two simultaneous connections are allowed.

Reply:
I'll split it.